mailhandler_d8.module

```diff
@@ -0,0 +1,35 @@
+<?php
+/**
+ * @file
+ * Hook implementations for Mailhandler module.
+ */
+
+use Drupal\Core\Entity\EntityInterface;
+
+/**
+ * Implements hook_ENTITY_TYPE_presave().
+ */
+function mailhandler_d8_user_presave(EntityInterface $entity) {
+  /** @var \Drupal\user\UserInterface $entity */
+  if ($entity->hasField('mailhandler_gpg_key') && extension_loaded('gnupg')) {
+    /** @var \Drupal\user\UserInterface|null $original */
+    $original = $entity->original;
+    $gpg_key_field = $entity->get('mailhandler_gpg_key');
+
+    // Update the GPG keyring in case a new user has been created
+    // or the GPG key has been changed.
+    if (!$original || !$gpg_key_field->equals($original->get('mailhandler_gpg_key'))) {
+      // Initialize GnuPG resource.
+      $gpg = gnupg_init();
+
+      // @todo: Delete the old key from GPG keyring.
+      // gnupg_deletekey($gpg, $original->get('mailhandler_gpg_key')->fingerprint);
```

Comment:
It uses the system's keyring. Each imported public key gets a Key ID. In case a signature is verified, the user's fingerprint (with its data) is returned. By looking at the key info for the returned fingerprint we can get more information like name, email etc.